### ISSUE TITLE
Merge `Mechanics` folders into a single gallery of puzzle mechanics

### ISSUE THREAD

**Issue by kostmo** [swarm-game/swarm#1088]
Let's create a new toplevel scenarios category (i.e. a sibling of "Fun"/"Tutorials", etc.) to serve as a collection of demonstrations of basic building blocks for more complex challenges.  None of them need be fully-fleshed out scenarios with the obligation of polished maps and flavor text.  In fact, the more bare-bones, the better, for the purpose of copy-pasting into larger scenarios.

Things that may belong in this category:
* demonstration of "trapdoor": #976
* portal demonstration (a la [`teleport.yaml`](https://github.com/swarm-game/swarm/blob/main/data/scenarios/Challenges/teleport.yaml)/#406)
* minimal demo of [enclosure checking](https://github.com/swarm-game/swarm/blob/main/data/scenarios/Challenges/Ranching/_gated-paddock/enclosure-checking.sw) (from #835)
* invent different types of enemies/dynamic obstacles

**Comment by xsebek** [comment]
Good idea, this would give some of our regression tests a higher purpose. :smile:

**Comment by byorgey** [comment]
Note that we do have a `Mechanics` folder now.  There's just not much in it at the moment.
